pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -66,15 +73,27 @@ func (f *Factory) CreateIfMissing(ctx context.Context, datacenter string, machin
 	if err != nil {
 		return fmt.Errorf("checking for template: %v", err)
 	}
+	osFamily := machineConfig.Spec.OSFamily
+	resizeBRDisk := false
```

Comment:
why define a flag that is only used in one place instead of just checking the condition when you do the branching?